### DISCUSSION TITLE
Example of setting environment variables for codelldb

### DISCUSSION THREAD

**Discussion by dpo** [mfussenegger/nvim-dap#946] [General]
Hi,

I tried using the snippet https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#environment-variables for codelldb but I keep getting the a parse error `Could not parse launch configuration: invalid type: sequence, expected a map`.

I also tried
```lua
env = {
  var1 = "value1",
  var2 = "value2",
},
```
but that doesn't work either. I get the error `The "program" attribute is required for launch`. The program attribute is set correctly; it certainly works when I comment out the `env` section.

Could you please give an example of how to set environment variables for the debug run?

Thanks!

**Comment by dpo** [comment]
Are there some debug flags I can activate so I can see what is being passed to codelldb and, possibly, the json file that codelldb works with? Does my question pertain to this package or to codelldb itself? Thanks.

**Comment by croquemadame** [comment]
@dpo 

I slightly modified the syntax to:

```lua
env = function()
           local variables = {}
           for k, v in pairs(vim.fn.environ()) do
                variables[string.format("%s",k)] = string.format("%s",v)
           end
           return variables
end,
```
and it worked for me.

**Comment by DrCheeseFace** [comment]
> Hi,
> 
> I tried using the snippet https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#environment-variables for codelldb but I keep getting the a parse error `Could not parse launch configuration: invalid type: sequence, expected a map`.
> 
> I also tried
> ```lua
> env = {
>   var1 = "value1",
>   var2 = "value2",
> },
> ```
> but that doesn't work either. I get the error `The "program" attribute is required for launch`. The program attribute is set correctly; it certainly works when I comment out the `env` section.
> 
> Could you please give an example of how to set environment variables for the debug run?
> 
> Thanks!

![20250712_113730.jpg](https://github.com/user-attachments/assets/063eeb18-a0af-443a-8736-c093dba3deee)

2 years too late but is this what you were looking for?
